development/cpp/index.rst

```diff
@@ -7,6 +7,7 @@ Engine development
 
    introduction_to_godot_development
    configuring_an_ide/index
+   using_cpp_profilers
```

Comment:
The current ordering of pages in this section is not the best, but this makes it worse IMO. Too many users skip reading documentation about the engine internals, and that's arguably something that one should look for before thinking about using a C++ profiler.

Not sure what would be a better toctree, but this might be worth splitting in several subsections (tooling, engine internals, extending the engine?).

Reply:
I reorganized the table of contents entirely, let me know what you think.